### PR TITLE
[OSDOCS-5331]: Adding docs about pausing reconciliation for hosted control planes

### PR DESCRIPTION
**Pull request by lahinson**
Version(s): 4.14

Issue: https://issues.redhat.com/browse/OSDOCS-5331

Link to docs preview: https://62528--docspreview.netlify.app/openshift-enterprise/latest/hosted_control_planes/hcp-managing.html#hosted-control-planes-pause-reconciliation_hcp-managing

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/hosted-control-planes-pause-reconciliation.adoc [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+. To pause reconciliation for a hosted cluster and hosted control plane, populate the `pausedUntil` field of the `HostedCluster` resource, as shown in the following example:
++
+[source,terminal]
+----
+PAUSED_UNTIL="2022-03-03T03:28:48Z"
+kubectl patch -n <hosted-clusters-namespace> hostedclusters/<hosted-cluster-name> -p '{"spec":{"pausedUntil":"'${PAUSED_UNTIL}'"}}' --type=merge
+----
++
+The value in the `pausedUntil` field can be a Boolean value, such as `PAUSED_UNTIL="true"` or an RFC339 timestamp, as shown in the example. If you specify an RFC339 date, the reconciliation is paused until that date. If you pass a Boolean value of `true`, reconciliation is paused until the field is removed from the `HostedCluster` resource.
```
Comment (bryan-cox) [inline modules/hosted-control-planes-pause-reconciliation.adoc:21]: nit: `If you specify an RFC339 date` should date be timestamp just to be consistent with the previous usage?

Comment (lahinson) [reply modules/hosted-control-planes-pause-reconciliation.adoc:21]: Yes. I'll change that. Good catch!

**Comment by bryan-cox** [comment]
> @bryan-cox Thanks again for your review. I have one question: In the upstream docs, the field is referred to as the `pausedUntil` field, but in the code example, it looks like `PAUSED_UNTIL`. I'm tempted to change the instances of `pausedUntil` to `PAUSED_UNTIL` for consistency. Is there any reason why I shouldn't do that?

The code example of it being all uppercase is just a normal practice of naming shell environment variables. 

[In the HyperShift API](https://github.com/openshift/hypershift/blob/e8938e1bfad2717d4d55e49009890522e7b55e03/api/v1beta1/hosted_controlplane.go#L135) it is named `PausedUntil`, but the CR representation is `pausedUntil` so I would say it's okay to leave it as is.

File: modules/hosted-control-planes-pause-reconciliation.adoc [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+.Procedure
+
+. To pause reconciliation for a hosted cluster and hosted control plane, populate the `pausedUntil` field of the `HostedCluster` resource, as shown in the following examples:
++
+** To pause the reconciliation until a specific time, specify an RFC339 timestamp:
++
+[source,terminal]
+----
+PAUSED_UNTIL="2022-03-03T03:28:48Z"
+kubectl patch -n <hosted-clusters-namespace> hostedclusters/<hosted-cluster-name> -p '{"spec":{"pausedUntil":"'${PAUSED_UNTIL}'"}}' --type=merge
```
Comment (muraee) [inline modules/hosted-control-planes-pause-reconciliation.adoc:20]:
```suggestion
kubectl patch -n <hosted-cluster-namespace> hostedclusters/<hosted-cluster-name> -p '{"spec":{"pausedUntil":"'${PAUSED_UNTIL}'"}}' --type=merge
```

File: modules/hosted-control-planes-pause-reconciliation.adoc [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+----
++
+The reconciliation is paused until the specified time is passed.
++
+** To pause the reconciliation indefinitely, pass a Boolean value of `true`:
++
+[source,terminal]
+----
+PAUSED_UNTIL="true"
+kubectl patch -n <hosted-clusters-namespace> hostedclusters/<hosted-cluster-name> -p '{"spec":{"pausedUntil":"'${PAUSED_UNTIL}'"}}' --type=merge
```
Comment (muraee) [inline modules/hosted-control-planes-pause-reconciliation.adoc:30]:
```suggestion
kubectl patch -n <hosted-cluster-namespace> hostedclusters/<hosted-cluster-name> -p '{"spec":{"pausedUntil":"'${PAUSED_UNTIL}'"}}' --type=merge
```

File: modules/hosted-control-planes-pause-reconciliation.adoc [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
++
+The reconciliation is paused until you remove the field from the `HostedCluster` resource.
++
+When the pause reconciliation field is populated for the `HostedCluster` resource, the field is automatically added to the associated `HostedControlPlane` resource.
+
+. To remove the `pausedUntil` field, enter the following patch command:
++
+[source,terminal]
+----
+kubectl patch -n <hosted-clusters-namespace> hostedclusters/<hosted-cluster-name> -p '{"spec":{"pausedUntil":null}}' --type=merge
```
Comment (muraee) [inline modules/hosted-control-planes-pause-reconciliation.adoc:41]:
```suggestion
kubectl patch -n <hosted-cluster-namespace> hostedclusters/<hosted-cluster-name> -p '{"spec":{"pausedUntil":null}}' --type=merge
```

File: modules/hosted-control-planes-pause-reconciliation.adoc (hunk 2 above)
Comment (lahinson) [reply modules/hosted-control-planes-pause-reconciliation.adoc:20]: Fixed! Good catch.

**Review by skrthomas — APPROVED** [review]
/lgtm

**Pull request merged**